skellytracker/trackers/mediapipe_tracker/mediapipe_observation.py

```diff
@@ -0,0 +1,140 @@
+import json
+from dataclasses import dataclass
+from typing import NamedTuple
+
+import numpy as np
+from mediapipe.python.solutions import holistic as mp_holistic
+from mediapipe.python.solutions.face_mesh import FACEMESH_NUM_LANDMARKS_WITH_IRISES
+from mediapipe.framework.formats.landmark_pb2 import NormalizedLandmarkList
+
+from skellytracker.trackers.base_tracker.base_tracker import BaseObservation
+
+
+@dataclass
+class MediapipeObservation(BaseObservation):
+    pose_landmarks: NormalizedLandmarkList
+    right_hand_landmarks: NormalizedLandmarkList
+    left_hand_landmarks: NormalizedLandmarkList
+    face_landmarks: NormalizedLandmarkList
+
+    image_size: tuple[int, int]
+
+    @classmethod
+    def from_holistic_results(cls,
+                                  mediapipe_results: NamedTuple,
+                                  image_size: tuple[int, int]):
+        return cls(
+            pose_landmarks=mediapipe_results.pose_landmarks,
+            right_hand_landmarks=mediapipe_results.right_hand_landmarks,
+            left_hand_landmarks=mediapipe_results.left_hand_landmarks,
+            face_landmarks=mediapipe_results.face_landmarks,
+            image_size=image_size
+        )
+
+    @property
+    def charuco_empty(self):
+        return self.detected_charuco_corner_ids is None
+
+    @property
+    def aruco_empty(self):
+        return self.detected_aruco_marker_ids is None
+
+    @property
+    def body_landmark_names(self) -> list[str]:
+        return [landmark.name.lower() for landmark in mp_holistic.PoseLandmark]
+
+    @property
+    def hand_landmark_names(self) -> list[str]:
+        return [landmark.name.lower() for landmark in mp_holistic.HandLandmark]
+
+    @property
+    def num_body_points(self) -> int:
+        return len(self.body_landmark_names)
+
+    @property
+    def num_single_hand_points(self) -> int:
+        return len(self.hand_landmark_names)
+
+    @property
+    def num_face_points(self) -> int:
+        return FACEMESH_NUM_LANDMARKS_WITH_IRISES
+
+    @property
+    def num_total_points(self) -> int:
+        return self.num_body_points + (2 * self.num_single_hand_points) + self.num_face_points
+
+    @property
+    def pose_trajectories(self) -> np.ndarray[..., 3]:  # TODO: not sure how to type these array sizes, seems like it doesn't like the `self` reference
```

Comment:
Typing is hard with numpy arrays - I think there are pacakages that help with it, but nothing particularly venerable. 

I think there's a way to set up custom type annotations and validators in pydantic (or dataclasses) or something? Or just type hint them as 'np.ndarray' and then just do a shape validation on initialization or read?